grants_tagger/predict.py

```diff
@@ -19,57 +19,54 @@
 DEFAULT_TFIDF_SVM_PATH = os.path.join(FILEPATH, '../models/tfidf-svm-2020.05.2.pkl')
 DEFAULT_LABELBINARIZER_PATH = os.path.join(FILEPATH, '../models/label_binarizer.pkl')
 
-DEFAULT_SCIBERT = BertClassifier()
-DEFAULT_SCIBERT.load(DEFAULT_SCIBERT_PATH)
 
-with open(DEFAULT_TFIDF_SVM_PATH, "rb") as f:
-    DEFAULT_TFIDF_SVM = pickle.loads(f.read())
+def predict_tags(X, probabilities=False, threshold=0.5,
+                 scibert_path=DEFAULT_SCIBERT_PATH,
+                 tfidf_svm_path=DEFAULT_TFIDF_SVM_PATH,
+                 label_binarizer_path=DEFAULT_LABELBINARIZER_PATH):
+    scibert = BertClassifier()
+    scibert.load(scibert_path)
 
-with open(DEFAULT_LABELBINARIZER_PATH, "rb") as f:
-    DEFAULT_LABELBINARIZER = pickle.load(f)
+    with open(tfidf_svm_path, "rb") as f:
+        tfidf_svm = pickle.loads(f.read())
 
+    with open(label_binarizer_path, "rb") as f:
+        label_binarizer = pickle.load(f)
 
-def sort_tags_probs(tags, probs, threshold):
-    """
-    Args:
-        tags: list of science tags
-        probs: list of probabilities
-    Returns:
-        sorted_tags_probs: list of tuple (tag, prob) sorted in ascending order
-    """
-    data = zip(tags.tolist()+['------ THRESHOLD ------'], probs.tolist()+[threshold])
-    sorted_tags_probs = sorted(data, key = itemgetter(1), reverse=True)
-    return sorted_tags_probs
-
-def predict_tags(x, probabilities=False, threshold=0.5,
-                 model=[DEFAULT_SCIBERT, DEFAULT_TFIDF_SVM],
-                 label_binarizer=DEFAULT_LABELBINARIZER):
-    '''Input example text when running the .py file in the terminal to return predicted grant tags - use format:'''
-    if type(model) == list: # ensemble of models
-        Y_pred_probs = np.zeros((1, len(label_binarizer.classes_)))
-        for model_ in model:
-            Y_pred_probs_model = model_.predict_proba([x])
-            Y_pred_probs += Y_pred_probs_model
+    Y_pred_probs = np.zeros((len(X), len(label_binarizer.classes_)))
+    for model in [scibert, tfidf_svm]:
+        Y_pred_probs_model = model.predict_proba(X)
```

Comment:
Just a note to myself that this loops assumes that the labels in both models are completely aligned and mapped to the same classes, which means you used the same binarizer.

Reply:
good point, maybe a comment to add